kornia/filters/filter.py

```diff
@@ -71,16 +91,119 @@ def filter2D(input: torch.Tensor, kernel: torch.Tensor,
+    independently at each depth channel of the tensor. Before applying the
+    kernel, the function applies padding according to the specified mode so
+    that the output remains in the same shape.
+
+    Args:
+        input (torch.Tensor): the input tensor with shape of
+          :math:`(B, C, D, H, W)`.
+        kernel (torch.Tensor): the kernel to be convolved with the input
+          tensor. The kernel shape must be :math:`(1, kD, kH, kW)`.
+        border_type (str): the padding mode to be applied before convolving.
+          The expected modes are: ``'constant'``,
+          ``'replicate'`` or ``'circular'``. Default: ``'replicate'``.
+        normalized (bool): If True, kernel will be L1 normalized.
+
+    Return:
+        torch.Tensor: the convolved tensor of same size and numbers of channels
+        as the input with shape :math:`(B, C, D, H, W)`.
+
+    Example:
+        >>> input = torch.tensor([[[
+        ...    [[0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.]],
+        ...
+        ...    [[0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 5., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.]],
+        ...
+        ...    [[0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.],
+        ...     [0., 0., 0., 0., 0.]]]])
+        >>> kernel = torch.ones(1, 3, 3, 3)
+        >>> kornia.filter3D(input, kernel)
+        torch.tensor([[[[[0., 0., 0., 0., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 0., 0., 0., 0.]],
+                        [[0., 0., 0., 0., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 0., 0., 0., 0.]],
+                        [[0., 0., 0., 0., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 5., 5., 5., 0.],
+                         [0., 0., 0., 0., 0.]]]])
+    """
+    testing.check_is_tensor(input)
+    testing.check_is_tensor(kernel)
+
+    if not isinstance(border_type, str):
+        raise TypeError("Input border_type is not string. Got {}"
+                        .format(type(kernel)))
+
+    if not len(input.shape) == 5:
+        raise ValueError("Invalid input shape, we expect BxCxDxHxW. Got: {}"
+                         .format(input.shape))
+
+    if not len(kernel.shape) == 4:
+        raise ValueError("Invalid kernel shape, we expect BxDxHxW. Got: {}"
+                         .format(kernel.shape))
+
+    borders_list: List[str] = ['constant', 'replicate', 'circular']
```

Comment:
This is a bit redundant to me. The value validation shall already be handled by ```torch.nn.functional.pad```.